client/rest-high-level/build.gradle

```diff
@@ -59,4 +59,17 @@ forbiddenApisMain {
   // core does not depend on the httpclient for compile so we add the signatures here. We don't add them for test as they are already
   // specified
   signaturesURLs += [PrecommitTasks.getResource('/forbidden/http-signatures.txt')]
+}
+
+javadoc {
+    dependsOn ':core:javadoc'
+    dependsOn ':client:rest:javadoc'
+    String host;
+    if (version.endsWith("-SNAPSHOT")) {
+        host = "https://snapshots.elastic.co";
+    } else {
+        host = "https://artifacts.elastic.co";
+    }
```

Comment:
Maybe we could use something like `ext.javadoc_host` (like `ext.bwc_tests_enabled` in elasticsearch build.gradle file) and reuse it in the sub projects.